include/aws/common/array_list.inl

```diff
@@ -220,15 +220,23 @@ int aws_array_list_get_at_ptr(const struct aws_array_list *AWS_RESTRICT list, vo
 
 AWS_STATIC_IMPL
 int aws_array_list_set_at(struct aws_array_list *AWS_RESTRICT list, const void *val, size_t index) {
-    size_t necessary_size = (index + 1) * list->item_size;
+    size_t necessary_size;
+    if(!aws_mul_size_checked((index + 1), list->item_size, &necessary_size)) {
+        return AWS_OP_ERR;
+    }
 
     if (list->current_size < necessary_size) {
         if (aws_array_list_ensure_capacity(list, index)) {
             return AWS_OP_ERR;
         }
     }
 
-    memcpy((void *)((uint8_t *)list->data + (list->item_size * index)), val, list->item_size);
+    size_t shift;
+    if(!aws_mul_size_checked(list->item_size, index, &shift)) {
+        return AWS_OP_ERR;
+    }
+
+    memcpy((void *)((uint8_t *)list->data + shift), val, list->item_size);
```

Comment:
Can this addition overflow?

Reply:
CBMC didn't report any overflows after the fix, but I'm not sure. Do you thing that would be the case because of the (uint8_t *) cast?